templates/_layout.html

```diff
@@ -47,6 +47,9 @@
 
     {% include "_header.html" %}
 
+    <script src="/static/js/modules/global-nav.js"></script>
```

Comment:
Does this need to be exactly here in HTML?

Ideally it should be moved to the bottom of the template where other scripts are - especially after raven is configured, so in case of any errors they would be reported to sentry.
Or even better integrated as a part of `base.js` not to be loaded separately.

If it is meant to run immediately then it probably should be called before `_header.html` is loaded (together with GA), not after it.

Reply:
> Does this need to be exactly here in HTML?

I moved it up so as to avoid the jumping of the page that you pointed out. The way to get the most consistent experience for the user is to run `canonicalGlobalNav.createNav` as high up in the page as possible - this way the global nav will always display, so nothing will be surprising.

Another other option would to write some custom styles to push the page down by default, sort of leave a placeholder for the global nav to be loaded into, and then load the global nav JS at the bottom of the page to load when the browser gets around to it. There are downsides to this approach:

- Although there'd be no "jumping", the user would still see a flicker as the global nav appears, which would lead to a slight feeling of the page not being entirely solid
- If the global nav failed to load for some reason (e.g. connection issues, JS error, the user has JS disabled) there'd be a weird gap above the header
- It takes control away from the global nav - if we wanted to, for example, make the global nav taller, then we'd have to remember to update snapcraft.io codebase

In general I like the idea of JS loading as soon as it's needed, so anything that is displayed immediately should be run ASAP. As you pointed out already, pages that visibly change after loading create a pretty shaky user experience. So I think I'll leave it at the top if that's okay with you Bartek.

> If it is meant to run immediately then it probably should be called before _header.html is loaded (together with GA), not after it.

I've moved it above the `_header`.

Reply:
Sure, if having it up the page solves the loading jumping it's fine to be there. Just wanted to make it consistently before header (which probably was the intention).